planner/core/find_best_task.go

```diff
@@ -870,11 +870,11 @@ func (ds *DataSource) isCoveringIndex(columns, indexColumns []*expression.Column
 
 // If there is a table reader which needs to keep order, we should append a pk to table scan.
 func (ts *PhysicalTableScan) appendExtraHandleCol(ds *DataSource) (*expression.Column, bool) {
-	handleCol := ds.handleCol
-	if handleCol != nil {
-		return handleCol, false
+	handleCols := ds.handleCols
+	if handleCols != nil {
```

Comment:
May this check be `false`? I think it's always `not nil`.

Reply:
handleCols may be set to nil in `DataSource.PruneColumns`.